ouroboros-consensus-cardano-tools/src/Cardano/Tools/DBAnalyser/Analysis.hs

```diff
@@ -574,251 +559,327 @@ traceLedgerProcessing
+    doCheck <- onlyCheckNumBlocks limit
+
+    initLedger' <- initialiseLedger ledgerDbFS cfg initLedger
+
+    (ldb, bs) <- ledgerDbAndBackingStore backing ledgerDbFS $ case initLedger of
+                                                                Left s -> Left (initLedger', s)
+                                                                Right v -> Right v
+
+    withFile mOutfile $ \outFileHandle -> do
+      let line = Builder.run $ DP.showHeaders separator
+                             <> separator
+                             <> "...era-specific stats"
+      Text.IO.hPutStrLn outFileHandle line
+
+      void $ runExceptT (consumeStream (mkStream doCheck GetBlock db) (castPoint . getTip $ initLedger') (ldb, 0) (process bs outFileHandle))
+      pure Nothing
+  where
+
+    lcfg = topLevelConfigLedger cfg
+    ccfg = topLevelConfigProtocol cfg
+
+    withFile :: Maybe FilePath -> (IO.Handle -> IO r) -> IO r
+    withFile (Just outfile) = IO.withFile outfile IO.WriteMode
+    withFile Nothing        = \f -> f IO.stdout
+
+    -- Separator for the data that is printed
+    separator = "\t"
+
+    process ::
+         LedgerBackingStore IO (ExtLedgerState blk)
+      -> IO.Handle
+      -> blk
+      -> (LedgerDB' blk, Word64)
+      -> IO (LedgerDB' blk, Word64)
+    process backingStore outFileHandle blk (ldb, sinceLast) = do
+        prevRtsStats <- GC.getRTSStats
+        let
+          -- Compute how many nanoseconds the mutator used from the last
+          -- recorded 'elapsedTime' till the end of the execution of the given
+          -- action. This function forces the evaluation of its argument's
+          -- result.
+          time act = do
+              tPrev <- GC.mutator_elapsed_ns <$> GC.getRTSStats
+              !r <- act
+              tNow <- GC.mutator_elapsed_ns <$> GC.getRTSStats
+              pure (r, tNow - tPrev)
+
+        let slot = blockSlot      blk
+        ((ldb', sinceLast'), tFlush) <- time $
+          if sinceLast == 100
+          then do
+            let (toFlush, toKeep) =
+                  ledgerDbFlush DbChangelogFlushAllImmutable ldb
+            Ouroboros.Consensus.Storage.LedgerDB.OnDisk.flush backingStore toFlush
+            pure (toKeep, 0)
+          else pure (ldb, sinceLast + 1)
+        let prevLedgerState = ledgerDbCurrent ldb'
+        -- We do not use strictness annotation on the resulting tuples since
+        -- 'time' takes care of forcing the evaluation of its argument's result.
+        (tkLdgrView, tForecast) <- time $ forecast            slot prevLedgerState
+        (tkHdrSt,    tHdrTick)  <- time $ tickTheHeaderState  slot prevLedgerState tkLdgrView
+        (hdrSt',     tHdrApp)   <- time $ applyTheHeader                           tkLdgrView tkHdrSt
+        let ks = getBlockKeySets blk
+        let aks = rewindTableKeySets (ledgerDbChangelog ldb') ks
+        (urs, tReadDb) <- time $ defaultReadKeySets (readKeySets backingStore) $ readDb aks
```

Comment:
Align with the other `(,) <- time` patterns. Possibly also align the `=` in let-bindings with the `<-` arrows? Might make the code slightly easier to read, though I haven't tried to see what it looks like